ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/ModuleImporter.java

```diff
@@ -0,0 +1,156 @@
+
+    public ModuleImporter() {
+        packageList = getPackagesFromDistRepo();
+    }
+
+    /**
+     * Check module availability in the distribution.
+     *
+     * @param module module name.
+     * @return availability of the module.
+     */
+    public boolean isModuleInDistRepo(String module) {
+        boolean isModuleInDistRepo = false;
+        String langModule = LANG + "." + module;
+        for (Package pkg : packageList) {
+            if (module.equals(pkg.packageName().value()) || langModule.equals(pkg.packageName().value())) {
+                isModuleInDistRepo = true;
+                break;
+            }
+        }
+
+        return isModuleInDistRepo;
+    }
+
+    /**
+     * Returns import statement.
+     *
+     * @param module input module.
+     * @return import statement.
+     */
+    public String getImportStatement(String module) {
+        String importStatement = "import ";
+        String langModule = LANG + "." + module;
+        String finalStatement = null;
+        for (Package pkg : packageList) {
+            if (pkg.packageName().value().equals(module)) {
+                finalStatement = importStatement + pkg.packageOrg().toString() + '/' + pkg.packageName().value();
+                break;
+            }
+            if (pkg.packageName().value().equals(langModule)) {
+                finalStatement = importStatement + pkg.packageOrg().toString() + '/' + LANG + "." + module;
+                break;
+            }
+        }
+
+        return finalStatement;
+    }
+
+    /**
+     * Return found undefined modules.
+     *
+     * @param diagnostics collection of evaluator diagnostics.
+     * @return distinct list of module errors.
+     */
+    public List<String> undefinedModules(Collection<Diagnostic> diagnostics) {
+        List<String> moduleErrors = new ArrayList<>();
+        for (Diagnostic diagnostic : diagnostics) {
+            //TODO update following statement using error code (BCE2000) issue : #33990
+            if (diagnostic.toString().contains(UNDEFINED_MODULE)) {
+                moduleErrors.add(diagnostic.toString().split("\n")[0].split(" ")[3]);
+            }
+        }
+
+        return moduleErrors.stream().distinct().collect(Collectors.toList());
+    }
+
+    /**
+     * Get packages available in the distributions.
+     */
+    private List<Package> getPackagesFromDistRepo() {
+        DefaultEnvironment environment = new DefaultEnvironment();
+        // Creating a Ballerina distribution instance
+        BallerinaDistribution ballerinaDistribution = BallerinaDistribution.from(environment);
+        PackageRepository packageRepository = ballerinaDistribution.packageRepository();
+        Map<String, List<String>> pkgMap = packageRepository.getPackages();
+
+        List<io.ballerina.projects.Package> packages = new ArrayList<>();
+        List<String> skippedLangLibs = Arrays.asList("lang.annotations", "lang.__internal", "lang.query");
```

Comment:
Why don't we keep this outside of the method? We could keep this as a `final` list right?

Reply:
Updated in e17fcf134c2456d78b8f384f5f92fc44b14b264d